build-tools/scripts/jenkinsfile

```diff
@@ -0,0 +1,484 @@
+            uploadArtifactsOnlyIfSuccessful: false, usploadZips: false, virtualPath: "${env.StorageVirtualPath}", storageType: 'blobstorage'
+        ])
+
+    } catch (error) {
+        echo "ERROR : publishPackages: Unexpected error: ${error}"
+        status = 1
+    }
+
+    return status
+}
+
+def cleanFiles(path) {
+    // path can include wildcard characters
+
+    def status = 0
+
+    try {
+
+        script {
+            sh """
+            # Note: Escape dollar \$ even in bash comments to avoid groovy templating errors; use non-escaped dollar to expand groovy variables
+            for file in `ls ${path}`; do
+                [ -f "\$file" ] || continue
+                echo "Cleaning file '\${file}''"
+                rm -f "\$file"
+            done
+            """
+        }
+
+    } catch (error) {
+        echo "ERROR : cleanFiles: Unexpected error: ${error}"
+        status = 1
+    }
+
+    return status
+}
+
+def cleanDirectories(path) {
+    // path can include wildcard characters
+
+    def status = 0
+
+    try {
+        sh """
+            # Note: Escape dollar \$ even in bash comments to avoid groovy templating errors; use non-escaped dollar to expand groovy variables
+            for dir in `ls -d ${path}`; do
+                [ -d "\$dir" ] || continue
+                echo "Cleaning directory '\${dir}''"
+                rm -rf "\$dir"
+            done
+        """
+    } catch (error) {
+        echo "ERROR : cleanDirectories: Unexpected error: ${error}"
+        status = 1
+    }
+
+    return status
+}
+
+pipeline {
+    
+    agent {
+        label "${botLabel}"
+    }
+
+    options {
+        timestamps()
+    }
+
+    stages {
+        stage('init') {
+            steps {
+                script {
+                    echo "GIT_BRANCH: ${env.GIT_BRANCH}"
+                    echo "CHANGE_TARGET: ${env.CHANGE_TARGET}"
+
+                    publishBuildFilePaths = "xamarin-android/xamarin.android-oss*.zip,xamarin-android/bin/${env.BuildFlavor}/bundle-*.zip,xamarin-android/bin/Build*/Xamarin.Android.Sdk*.vsix,xamarin-android/prepare-image-dependencies.sh,xamarin-android/build-status*,xamarin-android/xa-build-status*";
+                    echo "init: publishBuildFilePaths: ${publishBuildFilePaths}"
+
+                    publishTestFilePaths = "xamarin-android/xa-test-errors*"
+                    echo "init: publishTestFilePaths: ${publishTestFilePaths}"
+
+                    echo "init: env.JOB_BASE_NAME: ${env.JOB_BASE_NAME}"
+                    if (env.JOB_BASE_NAME?.toLowerCase().startsWith("xamarin-android-pr")) {
+                        echo "Build type: PR"
+                        isPR = true
+                    } else {
+                        echo "Build type: CI"
+                    }
+                }
+
+                echo "Stage result: init: ${currentBuild.currentResult}"
+            }
+        }
+
+        stage('clean') {
```

Comment:
Why do we need a `clean` stage in the first place?  The `monodroid` repo doesn't have such a thing.

Reply:
Fixed with commit https://github.com/xamarin/xamarin-android/commit/aa03f1a92195a2860e0db9e89da349e50a941972
See comment in previous conversation